superset/connectors/sqla/models.py

```diff
@@ -906,9 +908,11 @@ def adhoc_column_to_sqla(
         """
         label = utils.get_column_name(col)
         expression = col["sqlExpression"]
+        sqla_metric = None
         if template_processor and expression:
             expression = template_processor.process_template(expression)
-        sqla_metric = literal_column(expression)
+        if expression and validate_adhoc_subquery(expression):
+            sqla_metric = literal_column(expression)
```

Comment:
Not sure if my explanation made sense, but my suggestion was to change this to:

```suggestion
        validate_adhoc_subquery(expression)
        sqla_metric = literal_column(expression)
```

Because when I read the code `if expression and validate_adhoc_subquery(expression):` I wonder "what happens when `validate_adhoc_subquery(expression)` is false?", but that's a situation that can never happen, because the function currently can only return `True` (or raise an exception).

Having the function return `None` instead makes it easier to understand what's going on where it's being used.

(Alternatively, you could modify the function to return `True` or `False`, and raise the exception at the call site based on its return value. But because you're calling it from multiple places I think it's not worth it.)